### PR TITLE
Add lowering of `aten.Int.Tensor` op.

### PR DESCRIPTION
**Pull request by pashu123**
The lowering of `aten.Int.Tensor` op has been added.
The changes has been made as a part of `convert-torch-to-linalg` pass.

### CODE REVIEW

**Comment by pashu123** [comment]
> > I was not able to find the respective torch python function that can emit `aten.Int.Tensor` op.
> 
> The python function that generates the `aten.Int.Tensor` op is `int(..)`. If you do something like
> 
> ```python
> def forward(self, x):
>     return int(x)
> ```
> 
> where `x` is a `torch` tensor, you get the following MLIR:
> 
> ```mlir
> func private @__torch__.MyModule.forward(%arg0: !torch.nn.Module<"__torch__.MyModule">, %arg1: !torch.tensor {torch.type_bound = !torch.vtensor<[],f32>}) -> !torch.int {
>     %1 = torch.operator "aten.Int.Tensor"(%arg1) : (!torch.tensor) -> !torch.int
>     return %1 : !torch.int
>   }
> ```
> 
> Although, I don't know if the `e2e_testing` currently supports returning ints. You could re-wrap the value in a tensor again before returning if this is an issue.

Yes, this seems to be an issue. The return type should be Memref type.

```
        Lowering Linalg-on-Tensors IR to LLVM with RefBackend failed with the following diagnostics:                                                                                  
        error: return value must be memref type                                                
        note: see current operation: "std.return"(%1) : (i64) -> () 
``` 
I can rewrap the integer into a tensor and return, but there is another op that pops up `"torch.aten.tensor.int"(%8, %6, %2, %5) : (!torch.int, !torch.int, !torch.none, !torch.bool) -> !torch.vtensor<[1],si64>` and its lowering is missing.

**Review by cathyzhyi — APPROVED** [review]

File: e2e_testing/torchscript/basic.py [hunk 1]
```diff
@@ -493,3 +493,22 @@ def forward(self, x):
...
+    @export
+    @annotate_args([
+        None,
+        ([], torch.int64, True),
+        ([], torch.float32, True),
+    ])
+    def forward(self, x, y):
+        # This is a workaround for not returning scalar value.
+        a =  int(x)
+        return y.add(y, alpha=a)
```
Comment (ramiro050) [inline e2e_testing/torchscript/basic.py:510]: this is pretty clever :)

Comment (pashu123) [reply e2e_testing/torchscript/basic.py:510]: @ramiro050  All thanks to @cathyzhyi.

Comment (cathyzhyi) [reply e2e_testing/torchscript/basic.py:510]: It was originally @silvasean 's idea!

**Pull request merged**